presidio-analyzer/analyzer/analyzer_engine.py

```diff
@@ -87,14 +91,24 @@ def __remove_duplicates(results):
 
         return filtered_results
 
+    @staticmethod
+    def __remove_low_scores(results, score_threshold):
```

Comment:
re-iterates the scores which is un-optimal. consider inject a predicate to the analyze method

Reply:
You mean to filter in advance? We'd like the entire list of results (prior to removal) to be traced